src/reader/mod.rs

```diff
@@ -236,17 +223,15 @@ impl InnerIndexReader {
     }
 
     fn reload(&self) -> crate::Result<()> {
-        let segment_readers = Self::open_segment_readers(&self.index)?;
-        let searcher_generation = Self::create_new_searcher_generation(
-            &segment_readers,
-            &self.searcher_generation_counter,
-            &self.searcher_generation_inventory,
-        );
+        let generation_id = 1 + self
```

Comment:
why +1 and fetch_add?

Reply:
[`fetch_add`](https://doc.rust-lang.org/stable/std/sync/atomic/struct.AtomicUsize.html#method.fetch_add) returns the previous value before doing the addition, hence the `+1` is necessary to obtain the result that is stored in the atomic.

(GCC for example provides both `__atomic_add_fetch` and `__atomic_fetch_add` which differ only in which value is returned, c.f https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html)